### DISCUSSION TITLE
How to add UTF-8 support in destination directory path?

### DISCUSSION THREAD

**Discussion by trprajapati161** [nmoinvaz/minizip#549] [Q&A]
How to add support of utf-8 in destination directory path ?

**Comment by nmoinvaz** [comment]
If you are using the _zip reader/writer api_, then the destination directory will be created for you by the writer and it will support UTF-8 if the zip entry has the UTF-8 flag set. 

If you are using the _zip api_ or the _compatibility layer api_ to extract to the disk then it will be up to you to create the destination directories and write the files. You are able to use the `mz_dir_make` function to create them which does support UTF-8. 

If you are experiencing a problem with either of these methods then you can open a bug report.

**Reply by trprajapati161** [reply]
I mean **int32_t mz_zip_reader_save_all(void *handle, const char *destination_dir)** does not support the unicode.

**Reply by nmoinvaz** [reply]
You have to convert the unicode path to UTF-8 and pass it in the destination dir.
